docker/daemon/client.go

```diff
@@ -0,0 +1,69 @@
+package daemon
+
+import (
+	"github.com/containers/image/types"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/tlsconfig"
+	"net/http"
+	"path/filepath"
+)
+
+const (
+	// The default API version to be used in case none is explicitly specified
+	defaultAPIVersion = "1.22"
+)
+
+// NewDockerClient initializes a new API client based on the passed SystemContext.
+func newDockerClient(ctx *types.SystemContext) (*dockerclient.Client, error) {
+	if ctx == nil {
+		return dockerclient.NewClient(dockerclient.DefaultDockerHost, defaultAPIVersion, nil, nil)
+	}
+
+	httpClient, err := tlsConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	host := ctx.DockerDaemonHost
+	if host == "" {
+		host = dockerclient.DefaultDockerHost
+	}
+
+	version := ctx.DockerDaemonAPIVersion
+	if version == "" {
+		version = defaultAPIVersion
+	}
+
+	cli, err := dockerclient.NewClient(host, version, httpClient, nil)
+	if err != nil {
+		return cli, err
+	}
+
+	return cli, nil
+}
+
+func tlsConfig(ctx *types.SystemContext) (*http.Client, error) {
+	if ctx == nil || ctx.DockerDaemonCertPath == "" {
```

Comment:
This means that `DockerDaemonInsecureSkipTLSVerify` does not work if `DockerDaemonCertPath` is not set.

Reply:
If there is no cert, there is also TLS, right? This is aligned with the implementation of _client.NewEnvClient_ and effectively is behaves now as if no context is specified. 

I think this behaviour is reasonable.

Reply:
No; the default behavior is to use the system-wide CA store to authenticate the server, which works fine without supplying certificates.

It’s the other way around AFAICS: because `DockerDaemonCertPath` fills all three values of `tlsconfig.Options`, once `DockerDaemonCertPath` is set, all three certificates must exist or `tlsconfig.Client` fails if I’m not mistaken. That would force a caller who just wants to disable TLS verification to provide three cert/key files, of which at least the CA one is completely useless. (Yeah, `NewEnvClient` seems to do it the same way; it still does not make sense.)

Reply:
fair enough, let me change this a bit